File: typescript/prime-factors/prime-factors.ts
export const calculatePrimeFactors = (start: number) => {

    let fac = []
    let divisor = 2

    while (start > 1) {
        while (start % divisor === 0) {
            fac.push(divisor);
            start /= divisor;
        }
        ++divisor;
    }

    return fac;


};

File: vbnet/prime-factors/PrimeFactors.vb
Imports System
Imports System.Collections.Generic

Public Module PrimeFactors
    Public Function Factors(ByVal start As Long) As Long()
        Dim longArr As List(Of Long) = New List(Of Long)()
        Dim divisor = 2L

        While start > 1
            While start Mod divisor = 0
                longArr.Add(divisor)
                start /= divisor
            End While
            divisor += 1
        End While
        Return longArr.ToArray()
    End Function
End Module
